biogeochem/EDLoggingMortalityMod.F90

```diff
@@ -418,6 +433,237 @@ end subroutine get_harvest_rate_area
+                                                 ! calculation here. Can think about optimizing the logging calculation
+                                                 ! in the future.
+      integer,  intent(out), optional :: cur_harvest_tag  ! harvest tag of the current cohort
+
+      ! Local Variables
+      integer :: h_index   ! for looping over harvest categories
+      integer :: icode   ! Integer equivalent of the event code (parameter file only allows reals)
+      real(r8) :: harvest_rate_c    ! Temporary variable
+      real(r8) :: harvest_rate_supply  ! Temporary variable
+
+     ! Loop around harvest categories to determine the hlm harvest rate demand and actual harvest rate for the 
+     ! current cohort based on patch history info
+     harvest_rate = 0._r8
+     harvest_rate_c = 0._r8
+     harvest_rate_supply = 0._r8
+     harvest_tag = 2
+
+     do h_index = 1,hlm_num_lu_harvest_cats
+        if (patch_anthro_disturbance_label .eq. primaryforest) then
+           if(hlm_harvest_catnames(h_index) .eq. "HARVEST_VH1") then! .or. &
+           !     hlm_harvest_catnames(h_index) .eq. "HARVEST_VH2") then
+              harvest_rate_c = harvest_rate_c + hlm_harvest_rates(h_index)
+              ! Determine the total supply of available C for harvest
+              if(harvestable_forest_c(h_index) >= harvest_rate_c) then
+                 harvest_rate_supply = harvest_rate_supply + harvestable_forest_c(h_index)
+                 harvest_tag(h_index) = 0
+              else
+                 harvest_tag(h_index) = 1
+              end if
+           endif
+        else if (patch_anthro_disturbance_label .eq. secondaryforest .and. &
+             secondary_age >= secondary_age_threshold) then
+           if(hlm_harvest_catnames(h_index) .eq. "HARVEST_SH1") then
+              harvest_rate_c = harvest_rate_c + hlm_harvest_rates(h_index)
+              if(harvestable_forest_c(h_index) >= harvest_rate_c) then
+                 harvest_rate_supply = harvest_rate_supply + harvestable_forest_c(h_index)
+                 harvest_tag(h_index) = 0
+              else
+                 harvest_tag(h_index) = 1
+              end if
+           endif
+        else if (patch_anthro_disturbance_label .eq. secondaryforest .and. &
+             secondary_age < secondary_age_threshold) then
+           if(hlm_harvest_catnames(h_index) .eq. "HARVEST_SH2") then! .or. &
+           !     hlm_harvest_catnames(h_index) .eq. "HARVEST_SH3") then
+              harvest_rate_c = harvest_rate_c + hlm_harvest_rates(h_index)
+              if(harvestable_forest_c(h_index) >= harvest_rate_c) then
+                 harvest_rate_supply = harvest_rate_supply + harvestable_forest_c(h_index)
+                 harvest_tag(h_index) = 0
+              else
+                 harvest_tag(h_index) = 1
+              end if
+           endif
+        endif
+     end do
```

Comment:
I'm not sure I understand the logic here of adding each category of`hlm_harvest_rates(:)` to the `harvest_rate_c` as you go through the loop, it seems like it should be only using the `hlm_harvest_rates(:)` indexed for each category within each patch type comparison?  I'm also unclear about the time normalization between `hlm_harvest_rates(:)` which is a rate and `harvestable_forest_c(:)` which is a stock?  Is there meant to be an implicit annualization in that comparison? If so,mshoul dit be made explicit?

Reply:
Yes for now. But since I will include non-forest harvest into the harvest rate, we now need these 3 categories to be separated. 

`hlm_harvest_rates` is in the unit of **kgC site-1 yr-1** and `harvestable_forest_c` is in the unit of **kgC site -1**. Thus the calculation here will get an annual `harvest_rate` in the unit of **area fraction yr-1**, by assuming the carbon fraction to be equivalent to the area fraction. I will add comments on these variables.